### PR TITLE
catalog backend: parse but do not store labels + annotations

### PR DESCRIPTION
**Pull request by freben**
Related to #880 and #767 

Looks large, but it's kinda mostly tests :) Now we should have a really solid foundation to stand on in terms of parsing and ingestion.

### CODE REVIEW

File: plugins/catalog-backend/src/ingestion/descriptors/ComponentDescriptorV1beta1Parser.ts [hunk 1]
```diff
@@ -16,48 +16,53 @@
...
+    });
+  }
 
-export class ComponentDescriptorV1Parser implements EnvelopeParser {
   async tryParse(
     envelope: DescriptorEnvelope,
   ): Promise<ParserOutput | undefined> {
     if (
-      envelope.apiVersion !== 'catalog.backstage.io/v1' ||
+      envelope.apiVersion !== 'backstage.io/v1beta1' ||
```
Comment (Rugvip) [inline plugins/catalog-backend/src/ingestion/descriptors/ComponentDescriptorV1beta1Parser.ts:53]: 👍 to dropping catalog prefix and adding beta1

File: plugins/catalog-backend/src/ingestion/descriptors/DescriptorEnvelopeParser.test.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import { unindent } from '../../util';
+import { makeValidator } from '../../validation';
+import { DescriptorEnvelopeParser } from './DescriptorEnvelopeParser';
+
+describe('DescriptorEnvelopeParser', () => {
+  let data: any;
+  let parser: DescriptorEnvelopeParser;
+
+  beforeEach(() => {
+    data = yaml.parse(unindent`
```
Comment (Rugvip) [inline plugins/catalog-backend/src/ingestion/descriptors/DescriptorEnvelopeParser.test.ts:27]: Do you even need to unindent this? think yaml usually supports leading whitespace

File: plugins/catalog-backend/src/ingestion/descriptors/types.ts [hunk 3]
```diff
@@ -15,16 +15,20 @@
  */
 
 import { ParserOutput } from '../types';
-import { DescriptorEnvelope } from './DescriptorEnvelope';
+import { DescriptorEnvelope } from './DescriptorEnvelopeParser';
 
-export type EnvelopeParser = {
+export type KindParser = {
```
Comment (Rugvip) [inline plugins/catalog-backend/src/ingestion/descriptors/types.ts:20]: 😊

File: plugins/catalog-backend/src/util/unindent.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+): string {
+  // Interweave the strings with the substitution vars first
+  let output = '';
+  for (let i = 0; i < values.length; i++) {
+    output += strings[i] + values[i];
+  }
+  output += strings[values.length];
+
+  // Split on newlines
+  const lines = output.split(/(?:\r\n|\n|\r)/);
```
Comment (Rugvip) [inline plugins/catalog-backend/src/util/unindent.ts:34]: `\r?\n`? `\r` isn't really a thing 😅

File: plugins/catalog-backend/src/validation/CommonValidatorFunctions.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+  /**
+   * Checks that the value can be safely transferred as JSON.
+   *
+   * @param value The value to check
+   */
+  static isJsonSafe(value: any): boolean {
+    try {
+      return lodash.isEqual(value, JSON.parse(JSON.stringify(value)));
+    } catch (e) {
```
Comment (Rugvip) [inline plugins/catalog-backend/src/validation/CommonValidatorFunctions.ts:61]: think we support `} catch {` here too

File: plugins/catalog-backend/src/validation/CommonValidatorFunctions.ts [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+  /**
+   * Normalizes by keeping only a-z, A-Z, and 0-9; and converts to lowercase.
+   *
+   * @param value The value to normalize
+   */
+  static normalizeToLowercaseAlphanum(value: string): string {
+    return value
+      .split('')
+      .filter((x) => /[a-zA-Z0-9]/.test(x))
```
Comment (Rugvip) [inline plugins/catalog-backend/src/validation/CommonValidatorFunctions.ts:104]: gotta run prettier here m8

**Review by Rugvip — APPROVED** [review]

File: plugins/catalog-backend/src/ingestion/descriptors/DescriptorEnvelopeParser.test.ts (hunk 2 above)
Comment (freben) [reply plugins/catalog-backend/src/ingestion/descriptors/DescriptorEnvelopeParser.test.ts:27]: Hahaha, did I do this unnecessarily?! Maybe I should have assumed less and just tried it...

File: plugins/catalog-backend/src/util/unindent.ts (hunk 4 above)
Comment (freben) [reply plugins/catalog-backend/src/util/unindent.ts:34]: Proudly copy pasted from the intarwebz! 😛

File: plugins/catalog-backend/src/validation/CommonValidatorFunctions.ts (hunk 6 above)
Comment (freben) [reply plugins/catalog-backend/src/validation/CommonValidatorFunctions.ts:104]: Whaaa i don't see it

File: plugins/catalog-backend/src/util/unindent.ts (hunk 4 above)
Comment (freben) [reply plugins/catalog-backend/src/util/unindent.ts:34]: Can't that be a thing on windows...

File: plugins/catalog-backend/src/validation/CommonValidatorFunctions.ts (hunk 6 above)
Comment (Rugvip) [reply plugins/catalog-backend/src/validation/CommonValidatorFunctions.ts:104]: we just changed the config back to not using paren around arrow params when possible xP

File: plugins/catalog-backend/src/util/unindent.ts (hunk 4 above)
Comment (Rugvip) [reply plugins/catalog-backend/src/util/unindent.ts:34]: nah, it's pre-osx Macs, windows is \r\n

File: plugins/catalog-backend/src/ingestion/descriptors/DescriptorEnvelopeParser.test.ts (hunk 2 above)
Comment (freben) [reply plugins/catalog-backend/src/ingestion/descriptors/DescriptorEnvelopeParser.test.ts:27]: removed...

**Pull request merged**